City-of-Helsinki/drupal-hdbt
# UHF-2498: Service channel layout

**NOTICE:** Still missing textmessage styles. Also we might need to tackle the issue with service channels without titles before merging this.

**Testing with platform:**

* If you don't already have a site for testing, create one:
  * `composer create-project City-of-Helsinki/drupal-helfi-platform:dev-main hel-platform --no-interaction --repository https://repository.drupal.hel.ninja/`
  * `cd hel-platform`
* Checkout correct branches: `composer require drupal/helfi_tpr:dev-UHF-2498_service_channel_layout && composer require drupal/hdbt:dev-UHF-2498_service_channel_layout`
* `make new`
* `make shell`
* Enable TPR module with configuration `drush en -y helfi_tpr_config`
* Import services `drush mim tpr_service`
* Import errand servives `drush mim tpr_errand_service`
* Import service channels `drush mim tpr_service_channel`
* `drush cr`
* `drush uli` and log into the site using the link.
* Go to view a services with different service channels. Here are some examples:
  * `fi//tpr-service/3641` (eservice, post, printable form)
  * `fi//tpr-service/6449` (eservice, website)
  * `fi//tpr-service/3005` (chat)
  * `fi/tpr-service/4641` (local)
  * `fi/tpr-service/2703` (eservice, email)
  * `fi/tpr-service/7629` (phone)
  * **MISSING (textmessage)**
* All the other channels should have white background with black borders style and eservice should be with secondary color as background color. Check that the layouts follow this design:  https://share.goabstract.com/2d4ad388-7753-4756-ba1d-6c6cbde6e425?collectionLayerId=3e29b4ca-53ff-49ab-a0a2-afc08a07270e&mode=build&selected=root-CEC4000B-8D0E-4EDE-AC72-7D780AC092E0

**Testing with existing instance:**

* Checkout correct branches: `composer require drupal/helfi_tpr:dev-UHF-2498_service_channel_layout && composer require drupal/hdbt:dev-UHF-2498_service_channel_layout`
* If you don't have any TPR data import it:
  * `make shell`
  * Enable TPR module with configuration `drush en -y helfi_tpr_config`
  * Import services `drush mim tpr_service`
  * Import errand servives `drush mim tpr_errand_service`
  * Import service channels `drush mim tpr_service_channel`
* `drush cr`
* Go to view a services with different service channels. Here are some examples:
  * `fi//tpr-service/3641` (eservice, post, printable form)
  * `fi//tpr-service/6449` (eservice, website)
  * `fi//tpr-service/3005` (chat)
  * `fi/tpr-service/4641` (local)
  * `fi/tpr-service/2703` (eservice, email)
  * `fi/tpr-service/7629` (phone)
  * **MISSING (textmessage)**
* All the other channels should have white background with black borders style and eservice should be with secondary color as background color. Check that the layouts follow this design:  https://share.goabstract.com/2d4ad388-7753-4756-ba1d-6c6cbde6e425?collectionLayerId=3e29b4ca-53ff-49ab-a0a2-afc08a07270e&mode=build&selected=root-CEC4000B-8D0E-4EDE-AC72-7D780AC092E0

Also check this:
https://github.com/City-of-Helsinki/drupal-module-helfi-tpr/pull/52
